packages/renderer/src/lib/ContainerList.svelte

```diff
@@ -40,18 +40,48 @@ import type { ContextUI } from './context/context';
 import Button from './ui/Button.svelte';
 import StateChange from './ui/StateChange.svelte';
 import SolidPodIcon from './images/SolidPodIcon.svelte';
+import Tab from './ui/Tab.svelte';
+import { findMatchInLeaves } from '../stores/search-util';
+import { Filter } from './container/filter';
 
 const containerUtils = new ContainerUtils();
 let openChoiceModal = false;
 let enginesList: EngineInfoUI[];
 
 // groups of containers that will be displayed
 let containerGroups: ContainerGroupInfoUI[] = [];
-let viewContributions: ViewInfoUI[] = [];
-let globalContext: ContextUI;
-let containersInfo: ContainerInfo[] = [];
-export let searchTerm = '';
-$: searchPattern.set(searchTerm);
+
+let tab: string;
+
+export let filter = '';
+let runningFilter: boolean;
+let stoppedFilter: boolean;
+
+$: filterObj = new Filter(decodeURI(filter));
+
+$: checkFilter(filterObj.rawFilter);
```

Comment:
if you don't use `filterObj.rawFilter` inside checkFilter maybe there is no reason to add it as a arg.
To trigger the call i think it also work something like `filterObj.rawFilter && checkfilter()`